x/bep3/genesis.go

```diff
@@ -0,0 +1,51 @@
+package bep3
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/bep3/types"
+)
+
+// InitGenesis initializes the store state from a genesis state.
+func InitGenesis(ctx sdk.Context, keeper Keeper, supplyKeeper types.SupplyKeeper, gs GenesisState) {
+	if err := gs.Validate(); err != nil {
+		panic(fmt.Sprintf("failed to validate %s genesis state: %s", ModuleName, err))
+	}
+
+	// Set each AssetSupply and store total coin count
+	totalAssetSupplyCoins := sdk.NewCoins()
+	for _, asset := range gs.AssetSupplies {
+		keeper.SetAssetSupply(ctx, asset, []byte(asset.Denom))
+		totalAssetSupplyCoins.Add(sdk.NewCoins(asset))
+	}
+
+	// Set each AtomicSwap and store total coin count
+	totalAtomicSwapCoins := sdk.NewCoins()
+	for _, swap := range gs.AtomicSwaps {
+		if swap.Validate() != nil {
+			panic(fmt.Sprintf("invalid swap %s", swap.GetSwapID()))
+		}
+		keeper.SetAtomicSwap(ctx, swap)
+		keeper.InsertIntoByBlockIndex(ctx, swap)
+		totalAtomicSwapCoins.Add(swap.GetModuleAccountCoins())
```

Comment:
see above

Reply:
Fixed in https://github.com/Kava-Labs/kava/pull/370/commits/aefaf6fdc9582bf6c83b94700b06889d27bdb39b